packages/flutter/lib/src/widgets/display_feature_sub_screen.dart

```diff
@@ -0,0 +1,233 @@
+  /// Creates a widget that positions its child so that it avoids display
+  /// features.
+  const DisplayFeatureSubScreen({
+    Key? key,
+    this.anchorPoint,
+    required this.child,
+  }) : super(key: key);
+
+  /// The anchor point used to pick the closest sub-screen.
+  ///
+  /// If the anchor point sits inside one of these sub-screens, then that
+  /// sub-screen is picked. If not, then the sub-screen with the closest edge to
+  /// the point is used.
+  ///
+  /// `Offset(0,0)` is the top-left corner of the available screen space. For
+  /// a dual-screen device, this is the top-left corner of the left screen.
+  final Offset? anchorPoint;
+
+  /// The widget below this widget in the tree.
+  ///
+  /// The padding on the [MediaQuery] for the [child] will be suitably adjusted
+  /// to zero out any sides that were avoided by this widget. The [MediaQuery]
+  /// for the [child] will no longer contain any display features that split the
+  /// screen into sub-screens.
+  ///
+  /// {@macro flutter.widgets.ProxyWidget.child}
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    assert(anchorPoint!=null || debugCheckHasDirectionality(
+        context,
+        why: 'to determine which sub-screen DisplayFeatureSubScreen uses',
+        alternative: "Alternatively, consider specifying the 'anchorPoint' argument on the DisplayFeatureSubScreen",
+    ));
+    final MediaQueryData mediaQuery = MediaQuery.of(context);
+    final Size parentSize = mediaQuery.size;
+    final Rect wantedBounds = Offset.zero & parentSize;
+    final Offset resolvedAnchorPoint = _finiteOffset(anchorPoint ?? _fallbackAnchorPoint(context));
+    final Iterable<Rect> subScreens = _subScreensInBounds(wantedBounds, _avoidBounds(mediaQuery));
+    final Rect closestSubScreen = _closestToAnchorPoint(subScreens, resolvedAnchorPoint);
+
+    return Padding(
+      padding: EdgeInsets.only(
+        left: closestSubScreen.left,
+        top: closestSubScreen.top,
+        right: parentSize.width - closestSubScreen.right,
+        bottom: parentSize.height - closestSubScreen.bottom,
+      ),
+      child: MediaQuery(
+        data: mediaQuery.removeDisplayFeatures(closestSubScreen),
+        child: child,
+      ),
+    );
+  }
+
+  static Offset _fallbackAnchorPoint(BuildContext context) {
+    final TextDirection textDirection = Directionality.of(context);
+    switch (textDirection) {
+      case TextDirection.rtl:
+        return const Offset(double.maxFinite, 0);
+      case TextDirection.ltr:
+        return Offset.zero;
+    }
+  }
+
+  static Iterable<Rect> _avoidBounds(MediaQueryData mediaQuery) {
+    return mediaQuery.displayFeatures.map((DisplayFeature d) => d.bounds)
+        .where((Rect r) => r.shortestSide > 0);
+  }
+
+  /// Returns the closest sub-screen to the [anchorPoint]
+  static Rect _closestToAnchorPoint(Iterable<Rect> subScreens, Offset anchorPoint) {
+    return subScreens.fold(subScreens.first,
```

Comment:
Same here, this may be a little simpler with a for loop and can also avoid re-computing the distance to the previous point.

Reply:
I agree. Changing to a simple loop